scripts/ioc_deploy.py

```diff
@@ -168,9 +219,83 @@ def main(args: CliArgs) -> int:
     if rval != ReturnCode.SUCCESS:
         logger.error(f"Nonzero return value {rval} from make")
         return rval
+    logger.info(f"Applying write protection to {deploy_dir}")
+    rval = set_permissions(
+        deploy_dir=deploy_dir, allow_write=False, dry_run=args.dry_run
+    )
+    if rval != ReturnCode.SUCCESS:
+        logger.error(f"Nonzero return value {rval} from set_permissions")
+        return rval
+    logger.info("ioc-deploy complete!")
     return ReturnCode.SUCCESS
 
 
+def main_perms(args: CliArgs) -> int:
+    """
+    All main steps of the only-apply-permissions script.
+
+    This will be called when args has at least one of apply_write_protection and remove_write_protection
+
+    Will either return an int code or raise.
+    """
+    logger.info("Running ioc-deploy write protection change: checking inputs")
+    if args.allow_write is None:
+        logger.error("Entered main_perms without args.apply_write selected")
+        return ReturnCode.EXCEPTION
+
+    deploy_dir = get_deploy_info(args).deploy_dir
+
+    if args.allow_write:
+        logger.info(f"Allowing writes to {deploy_dir}")
+    else:
+        logger.info(f"Preventing writes to {deploy_dir}")
+    if not args.auto_confirm:
+        user_text = input("Confirm target? y/n\n")
+        if not user_text.strip().lower().startswith("y"):
```

Comment:
Out of curiosity, why not re-use ```parse_allow_write``` here? The naming is maybe not ideal, but seems to be doing something very similar.

Reply:
You're right, I can re-use `parse_allow_writes` in all of the user confirmation sections. This and the other confirmation sections were written prior to me adding `parse_allow_writes` for the cli extension, there was a worse/different CLI handling in the previous iteration and I never went back to consider revisions. I'll make some changes for this.

Reply:
I'll also use this opportunity to consolidate the auto_confirm logic as per Robert's offline suggestion